core/src/main/java/org/mule/runtime/core/api/processor/strategy/ProcessingStrategy.java

```diff
@@ -22,11 +21,43 @@
  */
 public interface ProcessingStrategy {
 
-  void configureProcessors(List<Processor> processors, MessageProcessorChainBuilder chainBuilder);
+  /**
+   * Enrich {@link Processor} function by adding pre/post operators to implement processing strategy behaviour.
+   *
+   * @param pipeline pipeline instance.
+   * @param pipelineFunction pipeline function.
+   * @return enriched pipeline function/
+   */
+  default Function<Publisher<Event>, Publisher<Event>> onPipeline(Pipeline pipeline,
+                                                                  Function<Publisher<Event>, Publisher<Event>> pipelineFunction) {
+    return publisher -> from(publisher).transform(pipelineFunction);
+  }
 
-  default Function<Publisher<Event>, Publisher<Event>> onProcessor(Processor messageProcessor,
-                                                                   Function<Publisher<Event>, Publisher<Event>> publisherFunction) {
-    return publisher -> from(publisher).transform(publisherFunction);
+  /**
+   * Enrich {@link Processor} function by adding pre/post operators to implement processing strategy behaviour.
+   *
+   * @param pipeline pipeline instance.
+   * @param pipelineFunction pipeline function.
+   * @param messagingExceptionHandler exception handle to use.
+   * @return enriched pipeline function
+   */
+  default Function<Publisher<Event>, Publisher<Event>> onPipeline(Pipeline pipeline,
+                                                                  Function<Publisher<Event>, Publisher<Event>> pipelineFunction,
+                                                                  MessagingExceptionHandler messagingExceptionHandler) {
+    return onPipeline(pipeline, pipelineFunction);
```

Comment:
No, because this is a default implementation, but implementations that override this do.